### DISCUSSION TITLE
NativeFirebaseError: [storage/unauthorized] error

### DISCUSSION THREAD

**Discussion by arkanos** [invertase/react-native-firebase#6515] [Q&A]
Hi everyone,

I have been integrating the module into an Expo sdk v46 app with the eas build profile and native code.

I have followed the steps from [https://rnfirebase.io/#expo](https://rnfirebase.io/#expo) and completed all of them. So far so good.

Now, when trying to upload an image to one of the buckets in Firebase Storage, I am receiving the following permissions error:

```plain
NativeFirebaseError: [storage/unauthorized] User is not authorized to perform the desired action.
```

I guess, this has to do for sure with the Storage rules configuration, but I have been trying several configurations without avail.

So here's the one I think it should be correct (based on the assumption the module will take my **google-services.json** credentials):

```go
rules_version = '2';
service firebase.storage {
  match /b/{bucket}/o {
    match /{allPaths=**} {
      allow read, write: if request.auth != null;
    }
  }
}
```

Is the **request.auth** object filled with the credentials automatically? Or is this just for the Authentication service that Firebase also provides (which I'm not using)?

I have the feeling I'm missing something but not quite sure where.

(If remove the ``: if request.auth != null;`` bit it works like a charm, ofc, but I don't want everyone messing around with my bucket T_T)

Here's the code I use for handling the upload:

```javascript
const updateUser: SubmitHandler<any> = async (data) => {
        setIsLoading(true);
        try {
            if (user && selectedImage) {
                const userAvatarStorageRef = storage().ref(`/avatars/${user?._id}`);
                const uploadImage = await userAvatarStorageRef.putFile(selectedImage)
                    .catch(error => {
                        console.error(error);
                    });

                if (uploadImage) {
                    data.picture = await userAvatarStorageRef.getDownloadURL();
                    setSelectedImage(null);
                }
            }

            const updatedUserData = await updateUser(user!._id, data)
                .catch(error => {
                    if (error.message) {
                        setSubmitMesage(t(`update.validation.${error.response.data.message}`))
                        setIsLoading(false);
                    }
                });


            if (updatedUserData) {
                await queryClient.invalidateQueries(['getSelfUserProfile']);
                setIsLoading(false);
            }
        } catch (error) {
            console.log(error)
        }

    };
```

Any help would be very much appreciated.

**Comment by mikehardy** [comment]
I think this is a general firebase question and might get a better response from their semi-official support on stackoverflow (if it is not already answered there). I think request.auth is filled in for you auto-magically if you have an authenticated user otherwise it does not exist, but I would confirm on stackoverflow
